### PR TITLE
New naming scheme

### PR DESCRIPTION
**Pull request by vweevers**
Counterpart of https://github.com/prebuild/prebuildify/pull/27. Major.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -32,54 +33,105 @@ load.path = function (dir) {
...
+    } else if (tag.slice(0, 3) === 'abi') {
+      tags.abi = tag.slice(3)
+    } else if (tag.slice(0, 2) === 'uv') {
+      tags.uv = tag.slice(2)
+    } else if (tag.slice(0, 4) === 'armv') {
+      tags.armv = tag.slice(4)
+    } else if (tag === 'glibc' || tag === 'musl') {
+      tags.libc = tag
+    } else {
+      return
```
Comment (mafintosh) [inline index.js]: Shouldn’t this be continue?

**Comment by mafintosh** [comment]
Looking really good! See my one comment

File: index.js (hunk 1 above)
Comment (vweevers) [reply index.js]: I figured if there's a filename containing an unknown tag, the file should be ignored altogether. No?

Comment (mafintosh) [reply index.js]: I think it should just ignore the tag. Sounds more future proof to me

File: index.js [hunk 2]
```diff
@@ -32,54 +33,105 @@ load.path = function (dir) {
...
+  return tags.runtime === 'node' && tags.napi
+}
+
+function compareTags (runtime) {
+  // Precedence: non-agnostic runtime, abi over napi, then by specificity.
+  return function (a, b) {
+    if (a.runtime !== b.runtime) {
+      return a.runtime === runtime ? -1 : 1
+    } else if (a.abi !== b.abi) {
+      return a.abi ? -1 : 1
```
Comment (ralphtheninja) [inline index.js]: @vweevers Mind explaining this a bit?

Comment (vweevers) [reply index.js]: We want to pick a prebuild that most closely matches the runtime (environment).

By this point in the code, we have already filtered out prebuilds with non-matching tags. But let's say the current runtime is Electron. There could still be:

1. A prebuild specifically targeting Electron
2. A runtime-agnostic prebuild (suitable for both Node and Electron)

If (1) exists, it should sort first and win. We're assuming that if (1) exists, it must mean that (2), though agnostic for _most_ runtimes, is not suitable for this specific runtime.

The logic is the same for the ABI's. At this point in the code, we could have:

1. A prebuild specifically targeting the current ABI
2. An "ABI-agnostic" prebuild (i.e. N-API)

If (1) exists, it should sort first and win.

**Pull request merged**